webapp/src/components/sidebar_right/index.tsx

```diff
@@ -0,0 +1,259 @@
+    if (!gilist) {
+        return [];
+    }
+
+    return gilist.map((pr: Item) => {
+        return {sha: pr.sha, project_id: pr.project_id, iid: pr.iid};
+    });
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string, prevState: string) {
+    if (currentState !== targetState) {
+        return false;
+    }
+
+    if (currentState !== prevState) {
+        return true;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
+        if (prs[i].id !== prevPrs[i].id) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function mapPrsToDetails(prs: Item[], details: Item[]) {
+    if (!prs || !prs.length) {
+        return [];
+    }
+
+    return prs.map((pr) => {
+        let foundDetails;
+        if (details) {
+            foundDetails = details.find((prDetails) => {
+                return (pr.project_id === prDetails.project_id) && (pr.sha === prDetails.sha);
+            });
+        }
+        if (!foundDetails) {
+            return pr;
+        }
+
+        return {
+            ...pr,
+            status: foundDetails.status,
+            approvers: foundDetails.approvers,
+            total_reviewers: pr.reviewers.length,
+        };
+    });
+}
+
+function SidebarRight({theme}: {theme: Theme}) {
+    const {username, yourAssignments, org, unreads, gitlabURL, rhsState, reviews, yourPrs, reviewDetails, yourPrDetails} = useSelector((state) => {
+        const store = getPluginState(state);
+        return {
+            username: store.username,
+            reviews: store.reviews,
+            reviewDetails: store.reviewDetails,
+            yourPrs: store.yourPrs,
+            yourPrDetails: store.yourPrDetails,
+            yourAssignments: store.yourAssignments,
+            unreads: store.unreads,
+            org: store.organization,
+            gitlabURL: store.gitlabURL,
+            rhsState: store.rhsState,
+        };
+    });
+
+    const [updatedPrs, setUpdatedPrs] = useState<Item[]>(yourPrs);
+    const [RHSState, setRHSState] = useState('');
+    const [updatedReviews, setUpdatedReviews] = useState<Item[]>(reviews);
+
+    useEffect(() => {
+        setUpdatedReviews(mapPrsToDetails(reviews, reviewDetails));
+    }, [reviews, reviewDetails]);
+
+    useEffect(() => {
+        setUpdatedPrs(mapPrsToDetails(yourPrs, yourPrDetails));
+    }, [yourPrs, yourPrDetails]);
+
+    useEffect(() => {
+        setUpdatedPrs(yourPrs);
+        setRHSState(rhsState);
+        setUpdatedReviews(reviews);
+        if (yourPrs && rhsState === RHSStates.PRS) {
+            getYourPrDetails(
+                mapGitlabItemListToPrList(yourPrs),
+            );
+            if (reviews && rhsState === RHSStates.REVIEWS) {
+                getReviewDetails(
+                    mapGitlabItemListToPrList(reviews),
+                );
+            }
+        }
+    }, []);
```

Comment:
we should update the `updatedReviews` after fetching the details of the reviews. Are you sure this code is working fine?
Also I see you have not use `useDispatch` hook anywhere. How are you dispatching a action?

Reply:
dispatching action correctly now